cmd/common.go

```diff
@@ -81,7 +82,13 @@ func syncMain(filenames []string, dry bool, parallelism, delay int) error {
 		return err
 	}
 	// prepare to read the current state from Kong
-	config.Workspace = targetContent.Workspace
+	if workspace != "" {
+		print.DeletePrintf("Warning: Specified workspace '%v' is different from "+
+			"workspace '%v' found in state file(s)\n", workspace, targetContent.Workspace)
```

Comment:
bug: Even if the `--workspace` provides the same workspace as the one in the state file(s), this warning will still be printed.
The right thing to do here is to first check if targetContent.Workspace and workspace are not same and then override config.Workspace as necessary.

Reply:
> Specified workspace

That's generic. May be "Workspace via --workspace flag '%v' ..."

Reply:
Thanks, fixed. The only edge case now is the default workspace which can be expressed as `default` and as an empty string. Let me know if you want to handle that as well and I'll add logic for it.

Reply:
> The only edge case now is the default workspace which can be expressed as default and as an empty string
 
It is not an edge case as long as user doesn't mix the two. If they do mix, they are doing it wrong and we can correct it.